### ISSUE TITLE
Additions to the evaluationCascade are no longer bound to the form instance

### ISSUE THREAD

**Issue by MartijnR** [enketo/enketo-core#892]
**Describe the bug**
Additions to the evaluationCascade are no longer bound to the form instance

It's a regression in https://github.com/enketo/enketo-core/commit/827c1f30a62d7e8a9093f7c9b1dac7b2b54151b4 that I should have prepared a test for, so we would have noticed. Sorry about that.

**Comment by MartijnR** [comment]
A fix is to push the unchanged functions to the stack or go back to simply concatenate them as before. The challenge is to write a test.

**Comment by lognaturel** [comment]
Here's the high-level gist I'm getting:
- the evaluation cascade itself is considered an integration point that clients can push arbitrary expressions into
- those expressions are expected to be evaluated in the context of the main instance
- they're not being evaluated in that context

Sort of?

**Comment by lognaturel** [comment]
https://github.com/enketo/enketo-core/pull/893#issuecomment-1132075839 suggests @eyelidlessness has all the context necessary so I can  discuss with him. When reviewing the original code I remember briefly wondering what cascade additions were. It did not occur to me that they were a client-configurable thing and it feels like it's not  documented as such?

**Comment by MartijnR** [comment]
Yes, it's not documented. It's a way to run a custom application without having to fork Enketo Core (just Enketo Express). It's good this broke, as I'm getting the incentive I needed to write some more of these tests (that I had promised...). :sweat_smile: I don't think documenting such advanced customizations is feasible but at least the test will provide some clarity to somebody that looks that deeply at Enketo.
